catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/PhiroIfBrickTest.java

```diff
@@ -0,0 +1,110 @@
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils.checkIfBrickAtPositionShowsString;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils.onScriptList;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.SpinnerUtils.checkIfValuesAvailableInSpinnerOnBrick;
+
+@RunWith(AndroidJUnit4.class)
+public class PhiroIfBrickTest {
+	private int brickPosition;
+	private PhiroIfLogicBeginBrick ifBrick;
+	private Script script;
+
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+
+		brickPosition = 1;
+
+		ifBrick = new PhiroIfLogicBeginBrick();
+		IfLogicElseBrick ifLogicElseBrick = new IfLogicElseBrick(ifBrick);
+		IfLogicEndBrick ifLogicEndBrick = new IfLogicEndBrick(ifLogicElseBrick, ifBrick);
+		ifBrick.setIfElseBrick(ifLogicElseBrick);
+		ifBrick.setIfEndBrick(ifLogicEndBrick);
+		script = BrickTestUtils.createProjectAndGetStartScript("PhiroIfBrickTest");
+		script.addBrick(ifBrick);
+		script.addBrick(ifLogicElseBrick);
+		script.addBrick(ifLogicEndBrick);
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void testPhiroIfBrick() {
+
+		List<Integer> spinnerValuesResourceIds = Arrays.asList(
+				R.string.phiro_sensor_front_left,
+				R.string.phiro_sensor_front_right,
+				R.string.phiro_sensor_side_left,
+				R.string.phiro_sensor_side_right,
+				R.string.phiro_sensor_bottom_left,
+				R.string.phiro_sensor_bottom_right);
+		checkIfBrickAtPositionShowsString(0, R.string.brick_when_started);
+		checkIfBrickAtPositionShowsString(brickPosition, R.string.brick_phiro_sensor_begin);
+		checkIfValuesAvailableInSpinnerOnBrick(spinnerValuesResourceIds, R.id.brick_phiro_sensor_action_spinner, brickPosition);
+		checkIfBrickAtPositionShowsString(brickPosition + 1, R.string.brick_if_else);
+		checkIfBrickAtPositionShowsString(brickPosition + 2, R.string.brick_if_end);
+	}
+
+	@Test
+	public void testPhiroIfBrickDelete() {
+		onScriptList().atPosition(brickPosition).onChildView(withText(R.string.brick_phiro_sensor_begin)).perform(click());
+		onView(withText(R.string.brick_context_dialog_delete_brick))
+				.perform(click());
+		onView(withText(R.string.yes))
+				.perform(click());
+
+		onView(withId(R.string.brick_phiro_sensor_begin)).check(doesNotExist());
```

Comment:
use onData for all checks on brickList. 

the way you do it here could be flaky, or result in a test that still succeeds even if something is broken. eg. dialog isnt closed, the checks below would not complain